src/main/java/io/cdap/plugin/dynamo/sink/DynamoDBBatchSinkConfig.java

```diff
@@ -0,0 +1,322 @@
+    primaryKeyTypes = builder.primaryKeyTypes;
+    readCapacityUnits = builder.readCapacityUnits;
+    writeCapacityUnits = builder.writeCapacityUnits;
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public static Builder builder(DynamoDBBatchSinkConfig copy) {
+    Builder builder = new Builder();
+    builder.referenceName = copy.referenceName;
+    builder.accessKey = copy.getAccessKey();
+    builder.secretAccessKey = copy.getSecretAccessKey();
+    builder.regionId = copy.getRegionId();
+    builder.endpointUrl = copy.getEndpointUrl();
+    builder.tableName = copy.getTableName();
+    builder.primaryKeyFields = copy.getPrimaryKeyFields();
+    builder.primaryKeyTypes = copy.getPrimaryKeyTypes();
+    builder.readCapacityUnits = copy.getReadCapacityUnits();
+    builder.writeCapacityUnits = copy.getWriteCapacityUnits();
+    return builder;
+  }
+
+  public String getAccessKey() {
+    return accessKey;
+  }
+
+  public String getSecretAccessKey() {
+    return secretAccessKey;
+  }
+
+  @Nullable
+  public String getRegionId() {
+    return regionId;
+  }
+
+  @Nullable
+  public String getEndpointUrl() {
+    return endpointUrl;
+  }
+
+  public String getTableName() {
+    return tableName;
+  }
+
+  public String getPrimaryKeyFields() {
+    return primaryKeyFields;
+  }
+
+  public String getPrimaryKeyTypes() {
+    return primaryKeyTypes;
+  }
+
+  @Nullable
+  public String getReadCapacityUnits() {
+    return readCapacityUnits;
+  }
+
+  @Nullable
+  public String getWriteCapacityUnits() {
+    return writeCapacityUnits;
+  }
+
+  /**
+   * Validates whether the table name follows the DynamoDB naming rules and conventions or not.
+   */
+  public void validateTableName(FailureCollector failureCollector) {
+    int tableNameLength = tableName.length();
+    if (tableNameLength < 3 || tableNameLength > 255) {
+      failureCollector.addFailure(
+        String.format("Table name '%s' does not follow the DynamoDB naming rules.", tableName),
+        "Table name must be between 3 and 255 characters long.")
+        .withConfigProperty(TABLE_NAME);
+    }
+
+    String pattern = "^[a-zA-Z0-9_.-]+$";
+    Pattern patternObj = Pattern.compile(pattern);
+
+    if (!Strings.isNullOrEmpty(tableName)) {
+      Matcher matcher = patternObj.matcher(tableName);
+      if (!matcher.find()) {
+        failureCollector.addFailure(
+          String.format("Table name '%s' does not follow the DynamoDB naming rules.", tableName),
+          "Table names can contain only the following characters: 'a-z, A-Z, 0-9, underscore(_), " +
+            "dot(.) and dash(-)'.")
+          .withConfigProperty(TABLE_NAME);
+      }
+
+    }
+  }
+
+  /**
+   * Validates the partition and sort key provided by the user.
+   */
+  public void validatePrimaryKey(FailureCollector failureCollector, Schema inputSchema) {
+    Set<String> primaryKeyAttributes = Splitter.on(',').trimResults().withKeyValueSeparator(":")
```

Comment:
it would be better for getPrimaryKeyFields to do this and return a Set<String>. That way the logic happens only there and not here and in the plugin.